srv/src/main/java/com/sap/cap/sflight/processor/ApprovalHandler.java

```diff
@@ -0,0 +1,101 @@
+package com.sap.cap.sflight.processor;
+
+import cds.gen.travelservice.AcceptTravelContext;
+import cds.gen.travelservice.RejectTravelContext;
+import cds.gen.travelservice.Travel;
+import cds.gen.travelservice.TravelService_;
+import static cds.gen.travelservice.TravelService_.TRAVEL;
+import cds.gen.travelservice.Travel_;
+import com.sap.cds.ql.Update;
+import com.sap.cds.services.cds.CdsService;
+import com.sap.cds.services.draft.DraftService;
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.handler.annotations.After;
+import com.sap.cds.services.handler.annotations.Before;
+import com.sap.cds.services.handler.annotations.On;
+import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.persistence.PersistenceService;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@ServiceName(TravelService_.CDS_NAME)
+public class ApprovalHandler implements EventHandler {
+
+	private static final String REJECT_TRAVEL = "rejectTravel";
+	private static final String ACCEPT_TRAVEL = "acceptTravel";
```

Comment:
The entity is unfortunately not inferred from the event context yet, even if it would be technically possible, but it is not implemented and the generated event context doesn't contain a hint to that yet.

Technically it will work if you leave out the entity, but only because this means that by default you will register on all entities. If another entity also had an action called `acceptTravel` the handler would be called for both of these entities. So to be precise: It is more correct to keep the entity in the handler registration.

Reply:
OK. So I'll keep the entities for now. :)